hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1227,19 +1235,25 @@ def defer_recursion(thing, producer):
             hints = get_type_hints(thing)
             params = get_signature(thing).parameters
         except Exception:
-            return builds(thing)
+            params = {}
         kwargs = {}
         for k, p in params.items():
             if (
                 k in hints
                 and k != "return"
-                and p.default is not Parameter.empty
                 and p.kind in (Parameter.POSITIONAL_OR_KEYWORD, Parameter.KEYWORD_ONLY)
             ):
-                kwargs[k] = defer_recursion(
-                    hints[k],
-                    lambda: just(p.default) | _from_type(hints[k], recurse_guard),
-                )
+                kwargs[k] = from_type_guarded(hints[k])
+                if p.default is not Parameter.empty and kwargs[k] is not ...:
+                    kwargs[k] |= just(p.default)
```

Comment:
Question: `p.default` is (or should be) an instance of the hinted type. Is the `just(p.default)` added to ensure `p.default` is generated as an example, or could it be dropped?

Reply:
We'll want to keep it because that _also_ ensures that the minimal failing example will use the default value.

Reply:
Ah, and to get the minimal example right we need `kwargs[k] = just(p.default) | kwargs[k]`.  

Let's add a test that checks we shrink to the minimal example?  That's also important for performance in the recursive case.